src/test/java/tests/ActionsWithProductsTest.java

```diff
@@ -0,0 +1,139 @@
+        productsPage.buyKitClick();
+        Assert.assertTrue(productsPage.getKitInCartText().contains("Вместе дешевле"));
+    }
+
+    @Test
+    public void productActionDeleteFromCartTest() {
+        HomePage homePage = new HomePage(getDriver());
+        ProductsPage productsPage = new ProductsPage(getDriver());
+
+        homePage.openHomePage();
+        homePage.navigateToCategoriesFromMainPage();
+        homePage.navigateToLaptopCategory();
+        productsPage.putFirstProductInCart();
+        productsPage.clickToHeaderCartButton();
+        productsPage.addMenuForDeleteProductFromCartClick();
+        productsPage.clickToDeleteInCartButton();
+        Assert.assertTrue(productsPage.getEmptyCartText().contains("Корзина пуста"));
+    }
+
+    @Test
+    public void productActionWishListTest() {
+        HomePage homePage = new HomePage(getDriver());
+        LoginPage loginPage = new LoginPage(getDriver());
+        ProductsPage productsPage = new ProductsPage(getDriver());
+
+        homePage.openHomePage();
+        loginPage.enterToAccountClick();
+        loginPage.inputLoginField();
+        loginPage.inputPasswordField();
+        loginPage.enterButtonClick();
+        homePage.navigateToCategoriesFromMainPage();
+        homePage.navigateToLaptopCategory();
+        productsPage.clickToFirstProduct();
+        productsPage.clickToProductTitle();
+        productsPage.firstWishButtonClick();
+        homePage.ActiveHeaderWishListButtonClick();
+        Assert.assertTrue(productsPage.getProductNameInWishList().contains("Ноутбук"));
+    }
+
+    @Test
+    public void wishListNewOwnListCreationTest() {
+        HomePage homePage = new HomePage(getDriver());
+        LoginPage loginPage = new LoginPage(getDriver());
+        ProductsPage productsPage = new ProductsPage(getDriver());
+
+        homePage.openHomePage();
+        loginPage.enterToAccountClick();
+        loginPage.inputLoginField();
+        loginPage.inputPasswordField();
+        loginPage.enterButtonClick();
+        jseHomePage.executeScript("scroll(0, 250);");
+        loginPage.accountWishListClick();
+        loginPage.createNewWishListClick();
+        loginPage.nameOfNewWishList();
+        loginPage.addNewListOfWishesClick();
+        Assert.assertTrue(loginPage.getNameOfWishList().contains("Ноутбуки"));
+        homePage.navigateToCategoriesFromMainPage();
+        homePage.navigateToLaptopCategory();
+        productsPage.clickToFirstProduct();
+        productsPage.clickToProductTitle();
+        productsPage.firstWishButtonClick();
+        homePage.ActiveHeaderWishListButtonClick();
+        Assert.assertTrue(productsPage.getProductNameInWishList().contains("Ноутбук"));
+
+
+    }
+
+    @Test
+    public void outOfStackProductTest() {
+        HomePage homePage = new HomePage(getDriver());
+        ProductsPage productsPage = new ProductsPage(getDriver());
+        LoginPage loginPage = new LoginPage(getDriver());
```

Comment:
all necessary pages can be created once at the top of class and used in all tests

Reply:
changed